src/server/shell/default_configuration.cpp

```diff
@@ -104,7 +104,15 @@ auto mir::DefaultServerConfiguration::the_idle_handler() -> std::shared_ptr<msh:
 
             auto options = the_options();
             int const idle_timeout_seconds = options->get<int>(options::idle_timeout_opt);
-            if (idle_timeout_seconds > 0)
+            if (idle_timeout_seconds < 0)
+            {
+                fatal_error("Invalid %s value %d, must be > 0", options::idle_timeout_opt, idle_timeout_seconds);
```

Comment:
I'd probably use `throw mir::AbnormalExit(...` (but non-blocking as it is a stupid user-error path.

Reply:
Easy enough